sound/soc/sof/loader.c

```diff
@@ -202,19 +209,34 @@ static int load_modules(struct snd_sof_dev *sdev, const struct firmware *fw)
 	/* parse each module */
 	module = (struct snd_sof_mod_hdr *)((u8 *)(fw->data) + sizeof(*header));
 	remaining = fw->size - sizeof(*header);
+	/* check for wrap */
+	if (remaining > fw->size) {
```

Comment:
this looks wrong. We should first check that fw->size >= sizeof(*header).
I didn't spot any other issues.

Reply:
@plbossart I tried to do "compliant post condition test for unsigned subtraction", like here:
https://wiki.sei.cmu.edu/confluence/display/c/INT30-C.+Ensure+that+unsigned+integer+operations+do+not+wrap
looks little bit strange, but should work...

Reply:
Ack, I think it's indeed correct -- the comment is good though.

Reply:
Ah, thanks for teaching me something :-)

For the record this is what @juimonen used:

````c
void func(unsigned int ui_a, unsigned int ui_b) {
  unsigned int udiff = ui_a - ui_b;
  if (udiff > ui_a) {
    /* Handle error */
  }
  /* ... */
}
````